File: lib/test/cpp/collatz.cpp
#include <iostream>
long collatz(long x)
{
    if (x <= 0)
    {
        return 1;
    }
    if (x % 2 == 0)
    {
        return collatz(x / 2);
    }
    return collatz(3 * x + 1);
}
int main()
{
    std::cout << collatz(100'0000'0000'0000'0000) << std::endl;
    return 0;
}

File: lib/test/zig/src/collatz.zig
export fn collatz(n: i128) i128 {
    if (n <= 1) {
        return 1;
    }
    if (@rem(n, 2) == 0) {
        return collatz(@divExact(n, 2));
    } else {
        return collatz(3 * n + 1);
    }
}
const std = @import("std");

pub fn main() anyerror!void {
    const stdout = std.io.getStdOut().writer();
    try stdout.print("{}", .{collatz(1000000000000000000)});
}
